### PR TITLE
feat: make author name and email check patterns configurable

### PR DESCRIPTION
**Pull request by zerwes**
introduces new commit config options `author_name_pattern` and `author_email_pattern`

## Summary by CodeRabbit

* **New Features**
  * Added configurable author email and author name pattern validation for commits.
  * Added command-line options and environment variables for customizing these patterns.
  * Author email validation now defaults to requiring an email-like format.

* **Documentation**
  * Documented the new configuration options, defaults, environment variables, and CLI flags.

* **Chores**
  * Added Vim swap files to the ignored files list.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ Quick Start
 
     repos:
       - repo: https://github.com/commit-check/commit-check
-        rev: v2.6.0
+        rev: v2.10.2
```
Comment (shenxianpeng) [inline README.rst]:
```suggestion
        rev: v2.11.0
```

File: README.rst [hunk 2]
```diff
@@ -198,12 +198,17 @@ For one-off checks or CI/CD pipelines, you can configure via CLI arguments or en
     # In pre-commit hooks (.pre-commit-config.yaml)
     repos:
       - repo: https://github.com/commit-check/commit-check
-        rev: v2.6.0
+        rev: v2.10.2
```
Comment (shenxianpeng) [inline README.rst]:
```suggestion
        rev: v2.11.0
```

File: README.rst [hunk 3]
```diff
@@ -224,7 +229,7 @@ branch's configured upstream:
...
```
Comment (shenxianpeng) [inline README.rst]:
```suggestion
        rev: v2.11.0
```

File: commit_check/config_merger.py [hunk 4]
```diff
@@ -78,6 +78,8 @@ def get_default_config() -> dict[str, Any]:
             "require_signed_off_by": DEFAULT_BOOLEAN_RULES["require_signed_off_by"],
             "ignore_authors": [],
             "ai_attribution": DEFAULT_AI_ATTRIBUTION,
+            "author_email_pattern": "^.+@.+$",
+            "author_name_pattern": "",
```
[the remark this answers is not in the document]
Comment (shenxianpeng) [reply commit_check/config_merger.py:82]: like AI code review, here is a breaking change, it will override default value of author_name:
```
        regex=r"^[A-Za-zÀ-ÖØ-öø-ÿ\u0100-\u017F\u0180-\u024F ,.'\-]+$|.*(\[bot])",
```

**Comment by shenxianpeng** [comment]
Hi @zerwes, thank you for your PR. Could you please explain a little for your use case why it needs to add `author_name_pattern` and `author_email_pattern`? (I'm not opposed to adding this feature, but I'd just like to hear about some use cases)

Also, could you add appropriate test cases to make sure that the newly added features work as expected?

File: commit_check/rule_builder.py [hunk 5]
```diff
@@ -236,6 +242,19 @@ def _build_author_list_rule(
...
+        """Build author name or email validation rule."""
+        regex = self.commit_config.get(config_key, "").strip()
+
+        return ValidationRule(
+            check=catalog_entry.check,
+            regex=regex,
+            error=catalog_entry.error,
+            suggest=catalog_entry.suggest,
+        )
+
```
[the remark this answers is not in the document]
Comment (zerwes) [reply commit_check/rule_builder.py:259]: see commit 9b3cfef
```diff
--- a/commit_check/rule_builder.py
+++ b/commit_check/rule_builder.py
@@ -246,7 +246,9 @@ class RuleBuilder:
         self, catalog_entry: RuleCatalogEntry, config_key: str
     ) -> ValidationRule | None:
         """Build author name or email validation rule."""
-        regex = self.commit_config.get(config_key, "").strip()
+        regex = catalog_entry.regex
+        if self.commit_config.get(config_key, ""):
+            regex = self.commit_config.get(config_key, "").strip()
 
         return ValidationRule(
             check=catalog_entry.check,
```

**Comment by zerwes** [comment]
> Hi @zerwes, thank you for your PR. Could you please explain a little for your use case why it needs to add `author_name_pattern` and `author_email_pattern`? (I'm not opposed to adding this feature, but I'd just like to hear about some use cases)

Hello @shenxianpeng 
My intention is to have some restricted validation regarding author email and author name as a pre-commit check. We use pre-commit for a wide variety of tests. And as the request for the author name and email validation was made, I came around your commit-check, and realized it meets my requirements, but just was missing the configurable patterns.

> Also, could you add appropriate test cases to make sure that the newly added features work as expected?

Surely. As there are some validation tests for the author name and email defined, I considered additional test as superfluous. Can you be so kind and give me a hint what kind of test you are thinking about?

Greetings from Berlin

**Comment by shenxianpeng** [comment]
@zerwes Greetings from Vilnius! 👋

Could you please add the following tests? like: 

1. Tests for `author_email_pattern` (pass/fail cases, Config/CLI/Env support).
2. Tests for `author_name_pattern` (pass/fail cases, Config/CLI/Env support).
3. Maybe regression tests to ensure the default author-name validation still works when no custom pattern is configured.
4. Maybe precedence tests verifying CLI/Environment Variable overrides TOML configuration.

I'd recommend letting AI generate as comprehensive a test suite as possible, or even let it review the PR and identify any missing test cases or edge cases to make sure everything is well covered.

**Comment by shenxianpeng** [comment]
I've rebased it onto main (it needed the `README.rst → README.md` migration resolved), added test coverage — custom name/email patterns, the env-var mapping, and a regression guard that an empty pattern falls back to the built-in regex rather than disabling the check — and swapped the example regex for a more inclusive one (the previous `^[A-Z]+...$` would reject mononyms, non-ASCII names, apostrophes, and lowercase names). Getting this merged now. Thanks for the contribution!

**Review by shenxianpeng — APPROVED** [review]

**Comment by zerwes** [comment]
Thanks for the fixes and the merge!
Sorry, was quite busy, still had the TODOs on my backlog, but they slid down as things with higher prio dropped in.
